src/compiler/commandLineParser.ts

```diff
@@ -3077,7 +3077,7 @@ function parseJsonConfigFileContentWorker(
         return getPropFromRaw(prop, isString, "string");
     }
 
-    function getPropFromRaw<T>(prop: "files" | "include" | "exclude" | "references", validateElement: (value: unknown) => boolean, elementTypeName: string): PropOfRaw<T> {
+    function getPropFromRaw<T>(prop: "files" | "include" | "exclude" | "references", validateElement: (value: unknown) => value is T, elementTypeName: string): PropOfRaw<T> {

```

Comment:
This doesn't make the code any safer than it currently is but I feel like it surfaces the "unsafe" part (the predicate function) to the reader and that code ends up being less surprising (previously it relied on the explicit type parameter and the contextual type provided through the return annotation: [TS playground](https://www.typescriptlang.org/play?ts=5.1.0-dev.20230315#code/C4TwDgpgBACgTgezAeQGYCUCGB3APAFQD4oBeKOCTAEwQDsAbEKfAbQF0oAfKAIloWABaTHDiYQPLr36CwiMDwDcAKGVUIAY3ojoqAK60NwAJZ0oxgM4BlYHGO0A5gApgEAB7AAXFAMBrfti0AJTerh7mFlAWtvYOKsr6hiZmDhDA8EgAYogAtlh4RE7KUFBySN48qMb0EBaS3Dz2Wnrq9bzuza1SPBSoEBSGtTwANMVQAG6Y9MZUmK4AojU5ELReUE6T9HoQ3n4BwaTEAEYICDWYtKMlEEsrwPjgEABymMve0XaOyiGw8mj5BGIAG8xhRgHo4LQoCwAOQwtjKAC+qkSRlMUNSwCskA0FmyCDyOAAjEUSmUwBUqjU6t0mlsug0OvSIDxvt4MigMDhcB9YsCxho6NEJlNSFBMRz8YTsE5ycMIjZPg55TxeY4eEEVCUwRCoZtFFAAPSGqD9RBwJEogxolJpbGaPG5fIAJlJpXklOqQ1phmZbR4TJaLLZvyQ-25aoc-O1aV14rSkqdOFl8nllkVsRVkY1BuNUAQviRQA))